### ISSUE TITLE
Hangs on "Logging In" step indefinitely

### ISSUE THREAD

**Issue by mtimkovich** [JasonZigelbaum/jqbx-issues#120]
After hitting the "Log in to the Web App" button, the site goes to the "Logging in" page, but never actually logs in.

I've tried clearing my cookies and other site data, but that didn't remedy it.

The console has the error

```
Failed to load resource: https://api.spotify.com/v1/users/{username}/playlists?limit=50&offset=0 the server responded with a status of 502 ()
```

**Comment by JasonZigelbaum** [comment]
A handful of people have contacted me about this- so far I can't see any rhyme or reason to the behavior. Can you send me your Spotify URI so I can look into it (you can email it to me jason[at]thebestagency.com). 

Thanks for the log. The error you're sending me makes it look like it's an issue w/ Spotify's API since they return a 502 error when trying to fetch your playlists.  I know they've recently made some changes so I'm guessing this is an unexpected consequence... Do you use any other third party Spotify apps? If so, are you having any issues with those as well?

**Comment by mtimkovich** [comment]
Even though this issue is on Spotify's end, it would be a good idea to add a timeout if the Spotify login step fails.

**Comment by Roanmh** [comment]
I am experiencing this behavior on Andriod (Pixel XL, 8.1). The app had difficulty upon my first login and was resolved after revoking access to your application and re-doing the process. It should also be noted that I denied Spotify's permission for everything except my public profile.

I am experiencing it again without resolution after having logged into the OS X client. I have tried swiping away and restarting both JQBX.fm and Spotify without change. JQBX will also show a plain white screen for varying amounts of time, as much as indefinitely.

I am happy to collect debugging info for you with instruction.

**Comment by JasonZigelbaum** [comment]
This issue should be resolved with the latest updates across all apps. Please let me know if it persists.

**Comment by mynameiswolfy** [comment]
I'm getting a 503 on the same url in the console.

https://api.spotify.com/v1/users/[username]/playlists?limit=50&offset=0

{
  "error": {
    "status": 503,
    "message": "Service unavailable"
  }
}

**Comment by JasonZigelbaum** [comment]
Thanks for reporting. This is currently an Issue w/ Spotify's API. They are
having trouble for one reason or another. It happens occasionally and
should be resolved within a half hour or so.

**Comment by nathanmerrill** [comment]
I'm stuck on this as well, but with a different console log:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://api.spotify.com/v1/me. (Reason: CORS request did not succeed)

I'm on Firefox

**Comment by JasonZigelbaum** [comment]
Hey thanks for feedback. What version of Firefox are you running? Also have
you ever been able to log in successfully in the past?

**Comment by nathanmerrill** [comment]
Yes, this is a recent development.  I can't say exactly when, but it changed within the last 2 weeks.

Firefox 67.0.4 (64-bit).

I've also tried a hard reload, as well as clearing site data, to no avail.

**Comment by JasonZigelbaum** [comment]
Thanks. So I just checked this out on FF ver 67.0.4 on my mac and I'm not
seeing any logs or issues. The fact that it's a CORS issue w/ the header on
the Spotify domain makes me think this has to do w/ one of Spotify's
servers. I checked on my end and they look correct though so I'm not 100%
sure. Is it possible to send me a screenshot of your network tab / a
screenshot of the error being logged in the console?

**Comment by nathanmerrill** [comment]
I didn't find the /me in the network call, and so after additional investigation, I discovered it was an extension I was using (Privacy Badger).  They must have updated their rules or something, as it now blocks calls to spotify by default.

**Comment by JasonZigelbaum** [comment]
Ah okay, thanks for letting me know. I'll try it out and see if I can get
them to change their default policy. In the meanwhile if you can set a rule
I think that's the best course of action for now.

**Comment by miguelpduarte** [comment]
Commenting here for visibility, the issue with Privacy Badger still persists. I have sent in a report for "Did Privacy Badger break this site?" to hopefully help with this.
